pl/lessons/libraries/benchee.md

```diff
@@ -0,0 +1,335 @@
+})
+```
+
+W tym przykładzie używamy tylko jednej listy liczb całkowitych od 1 do 10,000. Zaktualizujmy to aby użyć kilku różnych wejść, dzięki czemu możemy zobaczyć, co się dzieje z mniejszymi i większymi listami. Otworzymy ten plik i zmienimy go w następujący sposób:
+
+```elixir
+map_fun = fn(i) -> [i, i * i] end
+inputs = %{
+  "small list" => Enum.to_list(1..100),
+  "medium list" => Enum.to_list(1..10_000),
+  "large list" => Enum.to_list(1..1_000_000)
+}
+
+Benchee.run(%{
+  "flat_map"    => fn(list) -> Enum.flat_map(list, map_fun) end,
+  "map.flatten" => fn(list) -> list |> Enum.map(map_fun) |> List.flatten end
+}, inputs: inputs)
+```
+
+Zauważysz dwie różnice. Najpierw mamy mapę `input` zawierającą informacje o naszych danych wejściowych. Przekazujemy tę mapę jako opcję konfiguracji do `Benchee.run / 2`.
+
+Ponieważ nasze funkcje wymagają argumentu, musimy zaktualizować nasze funkcje tak, aby przyjmowały argument:
+
+```elixir
+fn -> Enum.flat_map(list, map_fun) end
+```
+
+teraz mamy:
+
+```elixir
+fn(list) -> Enum.flat_map(list, map_fun) end
+```
+
+Uruchommy to ponownie:
+
+```shell
+$ mix run benchmark.exs
+```
+
+Teraz powinieneś zobaczy następujące dane w konsoli:
+
+```shell
+Operating System: macOS
+CPU Information: Intel(R) Core(TM) i5-4260U CPU @ 1.40GHz
+Number of Available Cores: 4
+Available memory: 8.589934592 GB
+Elixir 1.5.1
+Erlang 20.0
+Benchmark suite executing with the following configuration:
+warmup: 2.00 s
+time: 5.00 s
+parallel: 1
+inputs: large list, medium list, small list
+Estimated total run time: 2.10 min
+
+Benchmarking with input large list:
+Benchmarking flat_map...
+Benchmarking map.flatten...
+
+Benchmarking with input medium list:
+Benchmarking flat_map...
+Benchmarking map.flatten...
+
+Benchmarking with input small list:
+Benchmarking flat_map...
+Benchmarking map.flatten...
+
+
+##### With input large list #####
+Name                  ips        average  deviation         median
+flat_map             6.29      158.93 ms    ±19.87%      160.19 ms
+map.flatten          4.80      208.20 ms    ±23.89%      200.11 ms
+
+Comparison:
+flat_map             6.29
+map.flatten          4.80 - 1.31x slower
+
+##### With input medium list #####
+Name                  ips        average  deviation         median
+flat_map           1.34 K        0.75 ms    ±28.14%        0.65 ms
+map.flatten        0.87 K        1.15 ms    ±57.91%        1.04 ms
+
+Comparison:
+flat_map           1.34 K
+map.flatten        0.87 K - 1.55x slower
+
+##### With input small list #####
+Name                  ips        average  deviation         median
+flat_map         122.71 K        8.15 μs   ±378.78%        7.00 μs
+map.flatten       86.39 K       11.58 μs   ±680.56%       10.00 μs
+
+Comparison:
+flat_map         122.71 K
+map.flatten       86.39 K - 1.42x slower
+```
+
+Teraz możemy zobaczyć informacje o naszych benchmarkach pogrupowane według danych wejściowych. Ten prosty
```

Comment:
Some paragraphs starting from this one are using unnecessary line-breaks. Could you please reformat those according to the original English lesson?